optimum/exporters/onnx/base.py

```diff
@@ -338,6 +338,17 @@ def generate_dummy_inputs_for_validation(self, reference_model_inputs: Mapping[s
         """
         return reference_model_inputs
 
+    def output_names_for_validation(self, reference_output_names: List[str]) -> List[str]:
```

Comment:
I think the name is not explicit, is it? Maybe it should be rather something like `filter_output_names`?

Reply:
It maps the output names of the ONNX model to the output names of the reference model and returns the output names to use for validation (useful when the output names between the ONNX model and the reference model differ).

Reply:
Then I don't understand https://github.com/huggingface/optimum/blob/df602513d4452959b03f79fc5b2866201f7f3db5/optimum/exporters/onnx/model_configs.py#L700

I actually don't understand what `Maps the output names of the ONNX model to the output names of the reference model.` means. In one case, the method is identity, in the second one it just returns ["sample"]

Reply:
For a U-Net model, the output names of the original model is `"sample"` while for the ONNX model the output name is `"out_sample"`. `output_names_for_validation` allows to match the output names so we can compare both outputs.

Reply:
Could we have it more explicit? Like the `torch_to_onnx_input_map`, where we would explicitely see the map

```
{"out_sample": "sample"}
```

or something like this, otherwise I don't understand where the `maps` description comes in play